### PR TITLE
[sgen] Fix the clearing of bits in the gc descriptor bitmap when using weak fields.

### PR DESCRIPTION
**Pull request by vargaz**
Fixes https://github.com/mono/mono/issues/7378.

### CODE REVIEW

File: mono/metadata/class-init.c [hunk 1]
```diff
@@ -5408,7 +5408,7 @@ mono_class_compute_gc_descriptor (MonoClass *klass)
 			for (int pos = 0; pos < weak_bitmap_nbits; ++pos) {
 				if (weak_bitmap [pos / BITMAP_EL_SIZE] & ((gsize)1) << (pos % BITMAP_EL_SIZE)) {
```
Comment (lewurm) [inline mono/metadata/class-init.c]: while at it, let's fix that puppy too? how about using a variable?

Comment (luhenry) [reply mono/metadata/class-init.c]: +1

**Comment by kumpera** [comment]
Using macros/inline functions to get/set/test bits would be great.

**Comment by marek-safar** [comment]
@monojenkins backport 2018-02

**Comment by marek-safar** [comment]
@monojenkins backport 2017-12

**Comment by marek-safar** [comment]
@vargaz could you please manually backport

**Pull request merged**